test/e2e/scrapeconfig_test.go

```diff
@@ -448,6 +448,71 @@ func testScrapeConfigKubernetesNodeRole(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// testScrapeConfigKubernetesInvalidRole tests that invalid roles are rejected by the operator, as expected.
+func testScrapeConfigKubernetesInvalidRole(t *testing.T) {
```

Comment:
ideally we'd need a dedicated test for all ScrapeConfig validations (similar to `testPrometheusCRDValidation()`).

Reply:
Added a few test-cases, let me know if we can add more to it.